pkg/restapi/wallet/chapibridge/operations.go

```diff
@@ -215,9 +239,70 @@ func (o *Operation) RequestApplicationProfile(rw http.ResponseWriter, req *http.
 // Responses:
 //    default: genericError
 //    200: chapiResponse
-func (o *Operation) SendCHAPIRequest(rw http.ResponseWriter, req *http.Request) {
-	// TODO : to be implemented [#633]
-	common.WriteErrorResponsef(rw, logger, http.StatusNotImplemented, "To be implemented !")
+func (o *Operation) SendCHAPIRequest(rw http.ResponseWriter, req *http.Request) { // nolint: funlen,gocritic
+	request, err := prepareCHAPIRequest(req.Body)
+	if err != nil {
+		logutil.LogError(logger, commandName, SendCHAPIRequest, err.Error())
+		common.WriteErrorResponsef(rw, logger, http.StatusBadRequest, err.Error())
+
+		return
+	}
+
+	profile, err := o.store.GetProfileByUserID(request.Body.UserID)
+	if err != nil {
+		logutil.LogError(logger, commandName, SendCHAPIRequest, err.Error())
+		common.WriteErrorResponsef(rw, logger, http.StatusBadRequest, err.Error())
+
+		return
+	}
+
+	if profile.ConnectionID == "" {
+		logutil.LogError(logger, commandName, SendCHAPIRequest, noConnectionFoundErr)
+		common.WriteErrorResponsef(rw, logger, http.StatusInternalServerError, noConnectionFoundErr)
+
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), request.Body.Timeout)
+	defer cancel()
+
+	msgBytes, err := json.Marshal(map[string]interface{}{
+		"@id":   uuid.New().String(),
+		"@type": chapiRqstDIDCommMsgType,
+		"data":  request.Body.Request,
+	})
+	if err != nil {
+		logutil.LogError(logger, commandName, SendCHAPIRequest, err.Error())
+		common.WriteErrorResponsef(rw, logger, http.StatusInternalServerError, err.Error())
+
+		return
+	}
+
+	responseBytes, err := o.messenger.Send(msgBytes,
```

Comment:
If I understand correctly, this message of type `https://trustbloc.dev/chapi/1.0/request` will be received by the "Mobile Mediator".

Why isn't a `forward` message used to transport this chapiRequest payload to the mediator?

Reply:
Right, thanks.